### PR TITLE
LPS-190013 Replace aui:button with clay:button in Echo modules

### PR DESCRIPTION
**Pull request by Stephy6**
Motivation
=======
The motivation behind this PR stadirize the usages of buttons in the portal by using for clay:button or clay:icon.
https://liferay.atlassian.net/browse/LPS-190013
https://liferay.atlassian.net/browse/LPS-192603

Proposed Solution
========
The proposed solution consist in substitute the usages of aui:button and buttons and <button> for clay:button or clay:icon.

Steps-to-Reproduce
========
For the site-navigation-admin-web:
1. Go to Navigation -> Click (+) to create a Site Navigation -> Check that the modal works correctly and renders the correct buttons
2. Inside the Navigation Menu created -> add some elements to it -> check that the buttons in the modal work correctly
3. Make some modifications in the in the items -> check that Save button in the sidebar panel works fine

For the assets-taglib
1. Go to Blogs -> add a new blog -> in the new blogs entry categorization section check that the tags and categories selector work fine

### CODE REVIEW

**Comment by liferay-echo** [comment]
Thank you for submitting this PR. 👏

Make sure it follows the [Echo Pull Request Template](https://gist.github.com/liferay-echo/5a4455285ff1800f9da146816d85e67d#file-pull-request-template-md) if you haven't already done so. 

Learn more about it [here](https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2027455014/Echo+Pull+Request+Etiquette).

File: modules/apps/asset/asset-taglib/src/main/resources/META-INF/resources/asset_categories_selector/page.jsp [hunk 1]
```diff
@@ -73,12 +73,6 @@ List<Map<String, Object>> vocabularies = (List<Map<String, Object>>)data.get("vo
 								</div>
 							</div>
 						</div>
-
```
Comment (veroglez) [inline modules/apps/asset/asset-taglib/src/main/resources/META-INF/resources/asset_categories_selector/page.jsp:76]: this button is necessary. It's an button that is rendered before the js loads, and thus we avoid this jump:

https://github.com/liferay-echo/liferay-portal/assets/9701095/efcc614b-6f77-4e8b-8277-867085995e89

File: modules/apps/asset/asset-taglib/src/main/resources/META-INF/resources/asset_tags_selector/page.jsp [hunk 2]
```diff
@@ -53,10 +53,6 @@ List<Map<String, String>> selectedItems = (List<Map<String, String>>)data.get("s
 				</div>
 			</div>
 		</div>
-
-		<button class="btn btn-secondary" type="button">
```
Comment (veroglez) [inline modules/apps/asset/asset-taglib/src/main/resources/META-INF/resources/asset_tags_selector/page.jsp:57]: Same for this one

File: modules/apps/site-navigation/site-navigation-admin-web/src/main/resources/META-INF/resources/add_site_navigation_menu_item.jsp [hunk 3]
```diff
@@ -45,9 +45,18 @@ if (addURL == null) {
...
 	<aui:button-row cssClass="modal-footer position-fixed">
-		<aui:button name="addButton" type="submit" value='<%= type.equals("layout") ? "select" : "add" %>' />
+		<clay:button
+			id='<%= liferayPortletResponse.getNamespace() + "addButton" %>'
+			label='<%= type.equals("layout") ? "select" : "add" %>'
+			type="submit"
+		/>
 
-		<aui:button href="<%= redirect %>" type="cancel" />
+		<clay:link
```
Comment (veroglez) [inline modules/apps/site-navigation/site-navigation-admin-web/src/main/resources/META-INF/resources/add_site_navigation_menu_item.jsp:54]: This seems to be not working correctly. The cancel button should close the modal and redirect if necessary. Compared to master, `aui:button` is rendering a button and doing a `liferay.util.navigate` in the `onClick`:

master

Now what is being done with `ClayLink` is rendering a link with a specific `href`:

branch 


So we need to keep this behavior using `clay:button`.

**Comment by veroglez** [comment]
@Stephy6 I have left a couple of comments 🙂

File: modules/apps/asset/asset-taglib/src/main/resources/META-INF/resources/asset_categories_selector/page.jsp (hunk 1 above)
Comment (Stephy6) [reply modules/apps/asset/asset-taglib/src/main/resources/META-INF/resources/asset_categories_selector/page.jsp:76]: Ahh understood

File: modules/apps/asset/asset-taglib/src/main/resources/META-INF/resources/asset_tags_selector/page.jsp (hunk 2 above)
Comment (Stephy6) [reply modules/apps/asset/asset-taglib/src/main/resources/META-INF/resources/asset_tags_selector/page.jsp:57]: Ok, thank you

**Comment by veroglez** [comment]
ci:forward

**Comment by ealonso** [comment]
ci:forward

**Comment by alessandromalves** [comment]
Only two unique failures were found by the CI ([comparison with acceptance routine here](https://testray.liferay.com/home/-/testray/runs/compare?formDate=1692796895100&testrayRunIdA=2410258854&testrayRunIdB=2411034616)) and were not reproducible locally. The others are not related to this PR. See https://github.com/brianchandotcom/liferay-portal/pull/139617

**Comment by alessandromalves** [comment]
ci:close

**Pull request closed without merging**